android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -315,3 +325,48 @@ fun String.extractLogicalIdUuid() = this.substringAfter("/").substringBefore("/"
 fun Resource.addTags(tags: List<Coding>) {
   tags.forEach { this.meta.addTag(it) }
 }
+
+suspend fun Task.updateDependentTaskDueDate(defaultRepository: DefaultRepository): Task {
+  return apply {
+    val isUpcoming = status == Task.TaskStatus.REQUESTED
+    val isOverdue =
+      (status == Task.TaskStatus.READY || status == Task.TaskStatus.INPROGRESS) &&
+        executionPeriod.end.before(Date())
+    val isDue = status == Task.TaskStatus.INPROGRESS
+    if (!(isUpcoming || isOverdue || isDue) || partOf.isEmpty()) {
```

Comment:
I didn't see these vars used below, I'm not convinced they make this clearer, since I have to remember what they mean to understand this conditional. I think it's clearer to put more logic in the conditional, you can also put the easier to evaluate thing first, I also think it's easier to read negation pushed in

```suggestion
    if (partOf.isEmpty() || (status !in listOf(Task.TaskStatus.REQUESTED, Task.TaskStatus.INPROGRESS) && !isOverdue)) {
```

feel free to play around with the conditional, I think we can make it simpler